Mathlib/Geometry/RingedSpace/LocallyRingedSpace.lean

```diff
@@ -313,6 +302,167 @@ instance component_nontrivial (X : LocallyRingedSpace.{u}) (U : Opens X.carrier)
+      Y.presheaf.stalkSpecializes (specializes_of_eq <| hfg ▸ hxx' ▸ rfl) ≫ g.stalkMap x' := by
+  subst hfg
+  subst hxx'
+  simp
+
+@[reassoc]
+lemma stalkMap_congr_hom (f g : X ⟶ Y) (hfg : f = g) (x : X) :
+    f.stalkMap x = Y.presheaf.stalkSpecializes (specializes_of_eq <| hfg ▸ rfl) ≫
+      g.stalkMap x := by
+  subst hfg
+  simp
+
+@[reassoc]
+lemma stalkMap_congr_point {X Y : LocallyRingedSpace.{u}} (f : X ⟶ Y) (x x' : X) (hxx' : x = x') :
+    f.stalkMap x ≫ X.presheaf.stalkSpecializes (specializes_of_eq hxx'.symm) =
+      Y.presheaf.stalkSpecializes (specializes_of_eq <| hxx' ▸ rfl) ≫ f.stalkMap x' := by
+  subst hxx'
+  simp
+
+@[reassoc (attr := simp)]
+lemma stalkMap_hom_inv (e : X ≅ Y) (y : Y) :
+    e.hom.stalkMap (e.inv.val.base y) ≫ e.inv.stalkMap y =
+      Y.presheaf.stalkSpecializes (specializes_of_eq <| by simp) := by
+  rw [← stalkMap_comp, LocallyRingedSpace.stalkMap_congr_hom (e.inv ≫ e.hom) (𝟙 _) (by simp)]
+  simp
+
+@[simp]
+lemma stalkMap_hom_inv_apply (e : X ≅ Y) (y : Y) (z) :
+    e.inv.stalkMap y (e.hom.stalkMap (e.inv.val.base y) z) =
+      Y.presheaf.stalkSpecializes (specializes_of_eq <| by simp) z :=
+  DFunLike.congr_fun (stalkMap_hom_inv e y) z
+
+@[reassoc (attr := simp)]
+lemma stalkMap_inv_hom (e : X ≅ Y) (x : X) :
+    e.inv.stalkMap (e.hom.val.base x) ≫ e.hom.stalkMap x =
+      X.presheaf.stalkSpecializes (specializes_of_eq <| by simp) := by
+  rw [← stalkMap_comp, LocallyRingedSpace.stalkMap_congr_hom (e.hom ≫ e.inv) (𝟙 _) (by simp)]
+  simp
+
+@[simp]
+lemma stalkMap_inv_hom_apply (e : X ≅ Y) (x : X) (y) :
+    e.hom.stalkMap x (e.inv.stalkMap (e.hom.val.base x) y) =
+      X.presheaf.stalkSpecializes (specializes_of_eq <| by simp) y :=
+  DFunLike.congr_fun (stalkMap_inv_hom e x) y
+
+-- @[reassoc]
+-- lemma stalkMap_germ (U : Opens Y)
+--     (x : (Opens.map f.val.base).obj U) :
+--     Y.presheaf.germ ⟨f.val.base x.val, x.property⟩ ≫ f.stalkMap x.val =
+--       f.val.c.app (op U) ≫ X.presheaf.germ x :=
+--   PresheafedSpace.stalkMap_germ f.val U x
+
+-- lemma stalkMap_germ_apply (U : Opens Y) (x : (Opens.map f.val.base).obj U) (y) :
+--     f.stalkMap x.val (Y.presheaf.germ ⟨f.val.base x.val, x.property⟩ y) =
+--       X.presheaf.germ x (f.val.c.app (op U) y) :=
+--   PresheafedSpace.stalkMap_germ_apply f.val U x y
+
+-- @[reassoc (attr := simp)]
+-- lemma stalkMap_germ' (U : Opens Y) (x : X) (hx : f.val.base x ∈ U) :
+--     Y.presheaf.germ ⟨f.val.base x, hx⟩ ≫ f.stalkMap x =
+--       f.val.c.app (op U) ≫ X.presheaf.germ (U := (Opens.map f.val.base).obj U) ⟨x, hx⟩ :=
+--   PresheafedSpace.stalkMap_germ' f.val U x hx
+
+-- @[simp]
+-- lemma stalkMap_germ'_apply (U : Opens Y) (x : X) (hx : f.val.base x ∈ U) (y) :
+--     f.stalkMap x (Y.presheaf.germ ⟨f.val.base x, hx⟩ y) =
+--       X.presheaf.germ (U := (Opens.map f.val.base).obj U) ⟨x, hx⟩ (f.val.c.app (op U) y) :=
+--   PresheafedSpace.stalkMap_germ_apply f.val U ⟨x, hx⟩ y
```

Comment:
For the record, this is the culprit of the slow down.
Still trying to figure out why. This isn't particularly slow on my machine.

Reply:
(or maybe the last lemma)